### PR TITLE
Update labelling system to reduce duplication and cover more stacks

### PR DESCRIPTION
**Pull request by dhruvkb**
## Description

This PR
- adds labelling for more stacks than covered before (like documentation, catalog and mgmt)
- uses `github-script` instead of duplicating the step for every stack
- removes use of `actions-ecosystem/action-add-labels` (last updated in Oct 2021)
- uses updated stacks from #1933

https://github.com/actions/labeler suggested by @krysal is great but we already have a list of changesets in `filter.yml` based on stacks and we can prepend "🧱 stack:" to them more easily compared to maintaining a separate `labeller.yml` file.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by sarayourfriend — APPROVED** [review]
LGTM! Any ideas for how we could somehow consolidate the list of stacks (and other labels, maybe?) into a single place to be referenced by others, rather than needing to manually maintain the list everywhere? It'd be nice to have some kind of label configuration like this that could be use by various places without needing to spread or duplicate the information out:

```yaml
labels:
  priority:
    low:
      emoji: green square
      urgency: ...
    medium:
      emoji: yellow square
      urgency: ...
    # etc...

  stacks:
    catalog:
      filters:
        - catalog/*
        - docker/minio/*
    ingestion_server:
      filters:
        - ingestion_server/*
    infra:
      # no filters because it's in a different repository
    # etc...
```

I suppose if something like that _did_ exist, we'd need to use a template script or something to generate the resulting actions that depend on those, at least for the ones that aren't able to make web requests :thinking: 

Anyway, this PR definitely looks great, no need for changes here, just wondering if there's some way to prevent drift in these various locations in the future :slightly_smiling_face:

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -125,26 +125,26 @@ jobs:
...
+          # - list of stack filters: `.github/filters.yml`
+          # - list of stack labels: https://github.com/WordPress/openverse/labels?q=stack
+          script: |
+            const stacks = ["catalog", "api", "ingestion_server", "frontend", "documentation", "mgmt"]
+            const labels = JSON
+              .parse('${{ needs.get-changes.outputs.changes }}')
+              .filter(change => stacks.includes(change))
+              .map(change => `🧱 stack: ${change.replace("_", " ")}`)
+            if (labels.len == 0) {
+              labels.push("🚦 status: awaiting triage", "🏷 status: label work required")
```
Comment (obulat) [inline .github/workflows/ci_cd.yml:140]: Is"label work required" a new label? I guess these two labels do describe different statuses, and we used just the "awaiting triage" for both.

**Review by obulat — APPROVED** [review]
LGTM!

File: .github/workflows/ci_cd.yml (hunk 1 above)
Comment (dhruvkb) [reply .github/workflows/ci_cd.yml:140]: It's not new, we've had that since the beginning.

**Pull request merged**